Check generator version on start up

oltionchampari opened the pull request:
Closes phovea/generator-phovea#194


I have added the `notifier.notify()` to the places where it is ensured that the message will  be printed
just once .

Since the generators compose with each other the message would be run multiple times .

![update-notfier](https://user-images.githubusercontent.com/51322092/67927268-73fc9c80-fbb8-11e9-8820-a4ee9d1c98d9.png)

I have tested every generator/subgenerator.
 
To test 

- lower the version of the generator-phovea on your local package.json,
- set `updateCheckInterval: 0` in *generators/_version/index.js* (see https://github.com/phovea/generator-phovea/pull/204/files#diff-9b7c1925207c623ed7d55e67b7af2827R15)
- run` yo phovea`

thinkh commented:
Check that the generator must be installed globally: `npm install -g generator-phovea`

oltionchampari commented:
> @oltionchampari Thanks for your work, it looks already good. I suggest to refactor the implementation to a separate generator command and compose it in to the base generators. The approach should work similar to the node/npm version check.

Refactored the implementation & and added the `-g` flag

thinkh commented:
Just for sake of documentation there is a minor "bug" when performing the following steps:

1. Lower the version of the generator-phovea on your local package.json,
2. Set `updateCheckInterval: 0` in *generators/_version/index.js* (see https://github.com/phovea/generator-phovea/pull/204/files#diff-9b7c1925207c623ed7d55e67b7af2827R15)
3. Run `yo phovea:_version` (or any other phovea command)
4. Reset version number in package.json (to current version)
5. Run `yo phovea:_version` (or any other phovea command)

**Result**
It will show the update message again, even though I have already the most recent version installed.

This behavior seems to be the opposite case of the one described in the [update-notifier documentation](https://github.com/yeoman/update-notifier#how):

> The first time the user runs your app, it will check for an update, and even if an update is available, it will wait the specified `updateCheckInterval` before notifying the user. This is done to not be annoying to the user, but might surprise you as an implementer if you're testing whether it works. Check out [`example.js`](example.js) to quickly test out `update-notifier` and see how you can test that it works in your app.

Since it seems to be complicated to fix it, I recommend to keep it like it is. In worst case the user gets the update notification twice before it's gone.
